metaseq/sequence_generator.py

```diff
@@ -212,24 +380,44 @@ def _generate(
         # first forward through all the fixed tokens with forced decoding we'll
         # need to handle normalization and prep for bookkeeping of incremental
         # decoding
-        start_step = src_tokens.shape[1]
+        start_step = src_tokens.shape[1]  # 30
```

Comment:
Does this indicate that we're forcing the num tokens to be 30? What about for GPT2-XL?